element: batch integration command when running a shell

abderrahim opened the pull request:
There is currently a huge overhead for for starting a buildbox-run
sandbox, we don't have to incur that overhead multiple times.

jjardon commented:
Build passes in my fork: https://github.com/jjardon/buildstream/actions/runs/462277574

jjardon commented:
CI in my fork: https://github.com/jjardon/buildstream/actions/runs/467143069

abderrahim committed:
element: batch integration command when running a shell

There is currently a huge overhead for for starting a buildbox-run
sandbox, we don't have to incur that overhead multiple times.

jjardon commented:
@cs-shadow This is for the bst-1 branch, not sure that PR covers that?

cs-shadow commented:
sorry, I overlooked that this PR was for `bst-1`. Seems like the new CI setup hasn't reached bst-1 yet so this would still be the right way to change things.

I'd be happy to merge this if you could please add fedora 32 as well so that we have some symmetry with the master branch.

juergbi reviewed (approved):
Thanks, this looks good to me.

buildbox-run should not have a big startup overhead, though, if you use buildbox-run-bubblewrap and buildbox-fuse.

abderrahim commented:
The startup overhead isn't that high indeed, it's the "shutdown" overhead (where it tries to find what changed in the sandbox) that is huge. (in the OP, I was referring to the entire overhead of buildbox-run compared to just running the commands, not specifically to the startup overhead).

juergbi commented:
> The startup overhead isn't that high indeed, it's the "shutdown" overhead (where it tries to find what changed in the sandbox) that is huge.

Good point, I hope I can get around to optimizing this: https://gitlab.com/BuildGrid/buildbox/buildbox-casd/-/issues/52
